component/loki/source/file/file.go

```diff
@@ -0,0 +1,277 @@
+package file
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log/level"
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/discovery"
+	"github.com/grafana/loki/clients/pkg/promtail/api"
+	"github.com/grafana/loki/clients/pkg/promtail/positions"
+	"github.com/prometheus/common/model"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name: "loki.source.file",
+		Args: Arguments{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+const (
+	pathLabel     = "__path__"
+	filenameLabel = "filename"
+)
+
+// Arguments holds values which are used to configure the loki.source.file
+// component.
+type Arguments struct {
+	Targets   []discovery.Target `river:"targets,attr,optional"`
+	ForwardTo []chan api.Entry   `river:"forward_to,attr,optional"`
+}
+
+// DefaultArguments defines the default settings for loki.source.file.
+var DefaultArguments = Arguments{}
+
+// UnmarshalRiver implements river.Unmarshaler.
+func (arg *Arguments) UnmarshalRiver(f func(interface{}) error) error {
+	*arg = DefaultArguments
+
+	type args Arguments
+	return f((*args)(arg))
+}
```

Comment:
nit: you can take this out since there's no defaults to apply. If we want to add defaults later, then we can add it back.